sdks/java/harness/src/main/java/org/apache/beam/fn/harness/FnHarness.java

```diff
@@ -203,7 +204,9 @@ public static void main(
       JvmInitializers.runBeforeProcessing(options);
 
       LOG.info("Entering instruction processing loop");
-      control.processInstructionRequests(options.as(GcsOptions.class).getExecutorService());
+      ExecutorService executorService = options.as(GcsOptions.class).getExecutorService();
+      control.processInstructionRequests(executorService);
+      executorService.shutdown();
```

Comment:
We should shutdown in finally to ensure shutdown is called.

Reply:
done